buffer/src/main/java/io/netty/buffer/PoolArena.java

```diff
@@ -375,7 +375,8 @@ int alignCapacity(int reqCapacity) {
     }
 
     void reallocate(PooledByteBuf<T> buf, int newCapacity, boolean freeOldMemory) {
-        if (newCapacity < 0 || newCapacity > buf.maxCapacity()) {
+        checkPositiveOrZero(newCapacity, "newCapacity");
```

Comment:
@thinkerou please revert this change... I would prefer to keep this as one if check.

Reply:
done, thanks!